lib/AzureSearchRequest.js

```diff
@@ -1,9 +1,21 @@
 const rp = require('request-promise-native');
 const azureSearchUrl = require('./AzureSearchUrl');
+const utils = require('./SearchApiVersionUtils');
+
+const isSupportedApiVersion = utils.isSupportedApiVersion;
+const getApiDefaultVersion = utils.getDefault;
+
+module.exports = async function azureSearchRequest(searchUrl, options = {}) {
+  const method = options.method || 'get';
+  const body = options.body || '{}';
+  const searchApiVersion = options.searchApiVersion || getApiDefaultVersion();
```

Comment:
Can the setting of `searchApiVersion` be done within `BlueGreenDeploymentOrchestrator/generator-function.js` e.g. https://github.com/nhsuk/apim-blue-green-deploy/pull/42/files#diff-1549f6b751204dd30e69dcc0c97cc431R15 as it means the setting is valid from the start of the functions execution. It would also prevent any part of the function running when it could have been prevented at the beginning of the execution.

Reply:
Had an internal debate about this when implementing and, on reflection, I think you are right. It's a concern of parameter validation in the orchestrator and not in the lib function.